com/company/Server.java

```diff
@@ -0,0 +1,137 @@
+package com.company;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.io.PrintWriter;
+import java.net.ServerSocket;
+import java.net.Socket;
+import java.nio.BufferUnderflowException;
+import java.util.ArrayList;
+import java.util.concurrent.Executor;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+
+public class Server implements Runnable{
+    private ArrayList<ConnectionHandler> connections;
+    private ServerSocket server;
+    private boolean done;
+    private ExecutorService pool;
+    public Server() {
+        this.connections = new ArrayList<>();
+        this.done = false;
+    }
+
+    @Override
+    public void run(){
+        try {
+            server = new ServerSocket(9999);
+            pool = Executors.newCachedThreadPool();
+            while(!done){
+                Socket client = server.accept();
+                ConnectionHandler handler = new ConnectionHandler(client);
+                connections.add(handler);
+                pool.execute( handler);
+
+            }
+        } catch (IOException e) {
+        shutdown();
+        }
+    }
+
+    public void broadcast (String message){
+        for (ConnectionHandler ch: connections){

```

Comment:
Here you broadcast the message to everyone including yourself. You can fix the problem by checking the nickname of the receiver.